framework/auth/core.py

```diff
@@ -843,6 +849,52 @@ def _get_unconfirmed_email_for_token(self, token):
 
         return verification['email']
 
+    def clean_email_verifications(self):
+        email_verifications = deepcopy(self.email_verifications)
+        for token in self.email_verifications:
+            try:
+                self.verify_token(token)
+            except (KeyError, ExpiredTokenError):
+                email_verifications.pop(token)
+        self.email_verifications = email_verifications
+        self.save()
+
+    def verified_email_get(self):
```

Comment:
Actually, `verified_emails` would also be a misnomer because this returns *un*verified emails, correct? So how about `unconfirmed_emails`? The variable in `get_globals` as well as the JS variable in contextVars should reflect this name change (e.g. `user_unconfirmed_emails` and `currentUser.unconfirmedEmails`).